paint/data/dataset.py

```diff
@@ -0,0 +1,437 @@
+            )
+        self.file_identifier = self._map_to_file_identifier(item_type)
+        self.root_dir = Path(root_dir)
+        if not self.root_dir.exists():
+            raise FileNotFoundError(
+                f"The root directory {self.root_dir} does not exist - you should download the dataset first!"
+            )
+
+        if item_ids is None:
+            item_ids = []
+            for item in self.root_dir.iterdir():
+                if item.is_file() and self.file_identifier in item.name:
+                    if self.file_identifier in [
+                        mappings.CALIBRATION_RAW_IMAGE_IDENTIFIER,
+                        mappings.CALIBRATION_CROPPED_IMAGE_IDENTIFIER,
+                        mappings.CALIBRATION_FLUX_IMAGE_IDENTIFIER,
+                        mappings.CALIBRATION_FLUX_CENTERED_IMAGE_IDENTIFIER,
+                    ]:
+                        item_ids.append(int(item.name.split("_")[0]))
+                    else:
+                        item_ids.append(int(item.name.split("-")[0]))
+        self.item_ids = item_ids
+        self.to_tensor = transforms.ToTensor()
+
+    def __str__(self) -> str:
+        """Generate a user-friendly representation of the dataset."""
+        return (
+            f"This is a dataset containing calibration items from the PAINT database:\n"
+            f"-The root directory is {self.root_dir}\n"
+            f"-The file identifier is {self.file_identifier}\n"
+            f"-The dataset contains {len(self.item_ids)} items\n"
+        )
+
+    @staticmethod
+    def _check_accepted_keys(key: str) -> None:
+        """
+        Check if the considered calibration item type is accepted.
+
+        Parameters
+        ----------
+        key : str
+            Key of the calibration item type to be checked.
+        """
+        accepted_items = [
+            mappings.CALIBRATION_RAW_IMAGE_KEY,
+            mappings.CALIBRATION_CROPPED_IMAGE_KEY,
+            mappings.CALIBRATION_FLUX_IMAGE_KEY,
+            mappings.CALIBRATION_FLUX_CENTERED_IMAGE_KEY,
+            mappings.CALIBRATION_PROPERTIES_KEY,
+        ]
+        if key not in accepted_items:
+            raise ValueError(
+                f"The calibration type {key} is not accepted. Please select one of "
+                f"{mappings.CALIBRATION_RAW_IMAGE_KEY}, {mappings.CALIBRATION_CROPPED_IMAGE_KEY}, "
+                f"{mappings.CALIBRATION_FLUX_IMAGE_KEY}, {mappings.CALIBRATION_FLUX_CENTERED_IMAGE_KEY}, or "
+                f"{mappings.CALIBRATION_PROPERTIES_KEY,}"
+            )
+
+    @staticmethod
+    def _map_to_file_identifier(key: str) -> str:
+        """
+        Convert a calibration item type to a calibration file identifier.
+
+        Parameters
+        ----------
+        key : str
+            Key to be mapped.
+
+        Returns
+        -------
+        str
+            Mapped key to the file identifier.
+        """
+        mapper = {
```

Comment:
This will be reconstructed over and over. Should this be a constant variable of the class instead?

Reply:
Good idea, II have changed this